File: Karaoke/Arquivo-em-Java/model/FilaKaraoke.java
import java.util.LinkedList;
import java.util.Queue;

public class FilaKaraoke {
    private Queue<Pessoa> fila;

    public FilaKaraoke() {
        fila = new LinkedList<>();
    }

    public void adicionarPessoa(Pessoa pessoa) {
        fila.add(pessoa);
    }

    public boolean removerPessoa(int id) {
        return fila.removeIf(p -> p.getId() == id);
    }

    public boolean alterarMusica(int id, String novaMusica) {
        for (Pessoa p : fila) {
            if (p.getId() == id) {
                p.setMusica(novaMusica);
                return true;
            }
        }
        return false;
    }

    public Pessoa chamarProxima() {
        return fila.poll();
    }

    public boolean isEmpty() {
        return fila.isEmpty();
    }

    public Queue<Pessoa> getFila() {
        return fila;
    }
}

File: Karaoke/Arquivo-em-Python/Model/fila_karaoke.py
class FilaKaraoke:
    def __init__(self):
        self.__fila = [] 

    def adicionar_pessoa(self, pessoa):
        self.__fila.append(pessoa)

    def remover_pessoa(self, id_busca):
        tamanho_inicial = len(self.__fila)
        self.__fila = [p for p in self.__fila if p.get_id() != id_busca]
        return len(self.__fila) < tamanho_inicial

    def alterar_musica(self, id_busca, nova_musica):
        for p in self.__fila:
            if p.get_id() == id_busca:
                p.set_musica(nova_musica)
                return True
        return False

    def chamar_proxima(self):
        if not self.is_empty():
            return self.__fila.pop(0)
        return False

    def is_empty(self):
        return len(self.__fila) == 0

    def get_fila(self):
        return self.__fila